lib/cylc/cycling/iso8601.py

```diff
@@ -56,15 +54,18 @@ class SuiteSpecifics(object):
     iso8601_parsers = None
 
 
-def memoize(function):
-    """This stores results for a given set of inputs to a function.
-
-    The inputs and results of the function must be immutable.
-    Keyword arguments are not allowed.
+def memoize(function, MEMOIZE_LIMIT=10000):
```

Comment:
I think it can be `memoize_limit` now, as it's not a constant anymore?

Reply:
Agreed, if kept as a function arg ... but I tend to think it should remain as a constant (either at module level, or internal to the `memoize` function).  This is used as a decorator, to modify the behavior of a bunch of methods, and do we have any reason to use different cache sizes in different methods at this stage??

Reply:
(Obviously that could be useful in principle, but at the moment we're only using the 10k size)

Reply:
(And once py3, we can look into functools lru cache too)

Reply:
I have moved it to inside the outer function. I could have put the number directly into the line `if len(inputs_results) > memoize_limit:`, if the value is to be stored inside this function, but I think setting it makes the fn more readable. 

It isn't used anywhere else so I don't think it should be global in scope.

No-one wants to set it from outside at the moment, so it doesn't need to be a kwarg.